includes/PerformanceFeature.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace NewfoldLabs\WP\Module\Performance;
+
+use NewfoldLabs\WP\ModuleLoader\Container;
+use NewfoldLabs\WP\Module\Performance\Performance;
+use NewfoldLabs\WP\Module\Performance\CacheTypes\Browser;
+use NewfoldLabs\WP\Module\Performance\CacheTypes\Cloudflare;
+use NewfoldLabs\WP\Module\Performance\CacheTypes\File;
+use NewfoldLabs\WP\Module\Performance\CacheTypes\Skip404;
+use NewfoldLabs\WP\Module\Performance\ResponseHeaderManager;
+
+use function NewfoldLabs\WP\Module\Performance\getCacheLevel;
+use function NewfoldLabs\WP\ModuleLoader\container as getContainer;
+
+/**
+ * Child class for a feature
+ *
+ * Child classes should define a name property as the feature name for all API calls. This name will be used in the registry.
+ * Child class naming convention is {FeatureName}Feature.
+ */
+class PerformanceFeature extends \NewfoldLabs\WP\Module\Features\Feature {
+
+	/**
+	 * The feature name.
+	 *
+	 * @var string
+	 */
+	protected $name = 'performance';
+
+	/**
+	 * The feature value. Defaults to on.
+	 *
+	 * @var boolean
+	 */
+	protected $value = true; // default to on
+
+	/**
+	 * Initialize performance feature.
+	 */
+	public function initialize() {
+		if ( function_exists( 'add_action' ) ) {
+
+			// Register module
+			add_action(
+				'plugins_loaded',
+				function () {
+					new Performance( getContainer() );
+				}
+			);
+
+			// Container Hooks
+			add_action(
+				'newfold_container_set',
+				function ( Container $container ) {
+					register_activation_hook(
+						$container->plugin()->file,
+						function () use ( $container ) {
+							Skip404::onActivation();
+							File::onActivation();
+							Browser::onActivation();
+							// Add headers to .htaccess
+							$responseHeaderManager = new ResponseHeaderManager();
+							$responseHeaderManager->addHeader( 'X-Newfold-Cache-Level', absint( getCacheLevel() ) );
+						}
+					);
+					register_deactivation_hook(
+						$container->plugin()->file,
+						function () use ( $container ) {
+							Skip404::onDeactivation();
+							File::onDeactivation();
+							Browser::onDeactivation();
+							// Remove all headers from .htaccess
+							$responseHeaderManager = new ResponseHeaderManager();
+							$responseHeaderManager->removeAllHeaders();
+						}
+					);
+				}
+			);
```

Comment:
As discussed, I don't think we'll want to conditionally run activation/deactivation hooks. I also don't think that these would run early enough if they are registered in an action hook callback.

Reply:
I've got these hooks integrated better and moved into the PerformanceFeatureHooks class.